orttraining/orttraining/python/training/ortmodule/__init__.py

```diff
@@ -11,7 +11,18 @@
 ################################################################################
 ONNX_OPSET_VERSION = 12
 MINIMUM_TORCH_VERSION_STR = '1.8.1'
-TORCH_CPP_BUILD_DIR = os.path.join(os.path.dirname(__file__),'torch_inline_extensions')
+
+# Check Torch CPP extension build directory
+home_dir = os.path.expanduser("~")
+python_package_dir = os.path.dirname(__file__)
+TORCH_CPP_BUILD_DIR = os.path.join(python_package_dir,'torch_inline_extensions')
+TORCH_CPP_BUILD_DIR_BACKUP = os.path.join(home_dir, '.cache', 'torch_ort_extensions')
+if not os.access(python_package_dir, os.X_OK | os.W_OK):
+    if os.access(home_dir, os.X_OK | os.W_OK):
+        TORCH_CPP_BUILD_DIR = TORCH_CPP_BUILD_DIR_BACKUP
+    else:
+        raise PermissionError('ORTModule could not find a writable directory to cache its internal files.',
```

Comment:
It may be possible that users are unable to make a directory writable. In that case, is it possible to not build the cpp extensions and turn off ```_use_external_gpu_allocator```?

Reply:
That is not the only CPP extension ORTModle supports, there is another for AtenOp support. In both cases and possible others future extensions, we could handle their disablement as an advanced configuration